File: space-game/Core.Tests/Extensions/StringExtensionsTests.vb
Imports NUnit.Framework
Imports Core.Extensions.StringExtensions

Namespace Extensions

    <TestFixture>
    Public Class StringExtensionsTests

        <TestCase("One", TestEnum.One)>
        <TestCase("two", TestEnum.Two)>
        <TestCase("three", TestEnum.Three)>
        Public Sub ToEnum_WhenCalledWithValidValue_ReturnsCorrectValue(testValue As String, expectedValue As TestEnum)
            Dim result = testValue.ToEnum(Of TestEnum)

            Assert.AreEqual(expectedValue, result)
        End Sub

        <TestCase("Four")>
        <TestCase("")>
        Public Sub ToEnum_WhenCalledWithInvalidValue_ReturnsNothing(value As String)
            Dim result = value.ToEnum(Of TestEnum)

            Assert.IsNull(result)
        End Sub

    End Class

    Public Enum TestEnum
        One
        Two
        Three
    End Enum
End Namespace

File: space-game/Core.Tests/Extensions/StringExtensionsTests.cs
using Core.Extensions;
using NUnit.Framework;

namespace Core.Tests.Extensions
{
    [TestFixture]
    public class StringExtensionsTests
    {
        [TestCase("One", TestEnum.One)]
        [TestCase("two", TestEnum.Two)]
        [TestCase("three", TestEnum.Three)]
        public void ToEnum_WhenCalledWithValidValue_ReturnsCorrectValue(string testValue, TestEnum expectedValue)
        {
            TestEnum? result = testValue.ToEnum<TestEnum>();

            Assert.IsNotNull(result);
            Assert.AreEqual(expectedValue, result.Value);
        }

        [TestCase("Four")]
        [TestCase("")]
        public void ToEnum_WhenCalledWithInvalidValue_ReturnsNothing(string value)
        {
            TestEnum? result = value.ToEnum<TestEnum>();

            Assert.IsNull(result);
        }

    }

    public enum TestEnum
    {
        One,
        Two,
        Three
    }
}
